### PR TITLE
Allow several font sizes for Subhead headings

### PR DESCRIPTION
**Pull request by camertron**
### What are you trying to accomplish?

This PR allows adjusting the (font) size of headings in the `Subhead` component. A Hubber asked in Slack why `Subhead` headings in PVC have a font size of 24px, but a font size of 20px in PRC and in Figma. Rather than change the font size and potentially mess with a bunch of existing usages in dotcom, I have chosen to introduce two heading sizes, large (the default), and medium.

### Screenshots

|Large|Medium|
|-|-|
|A screenshot of the Subhead component. Some heading text of 'My Heading' appears above a muted description of 'My Description'. The font size of the heading is large (24px).|A screenshot of the Subhead component. Some heading text of 'My Heading' appears above a muted description of 'My Description'. The font size of the heading is medium (20px).|

### Integration

No changes necessary in production.

#### Risk Assessment
  

- [x] **Low risk** the change is small, highly observable, and easily rolled back.
- [ ] **Medium risk** changes that are isolated, reduced in scope or could impact few users. The change will not impact library availability.
- [ ] **High risk** changes are those that could impact customers and SLOs, low or no test coverage, low observability, or slow to rollback.

### Accessibility

- **No new axe scan violation** - This change does not introduce any new [axe scan](https://thehub.github.com/epd/engineering/dev-practicals/frontend/accessibility/readiness-routine/development/#axe-scans) violations.

### Merge checklist

- [x] Added/updated tests
- [x] Added/updated documentation
- [x] Added/updated previews (Lookbook)
- [x] Tested in Chrome
- [x] Tested in Firefox
- [x] Tested in Safari
- [x] Tested in Edge

Take a look at the [What we look for in reviews](https://github.com/primer/react/blob/main/contributor-docs/CONTRIBUTING.md#what-we-look-for-in-reviews) section of the contributing guidelines for more information on how we review PRs.

### CODE REVIEW

**Comment by lesliecdubs** [comment]
Thanks for working on this, @camertron! 

Let's make sure to get an approval from someone on [Primer Design](https://github.com/github/design-infrastructure#primer-design) on this PR before merge.

File: app/components/primer/beta/subhead.pcss
Comment (jonrohan) [inline app/components/primer/beta/subhead.pcss:29]: I wonder if we have primitives for these values? @langermank

**Review by jonrohan — APPROVED** [review]
Code changes look good 👍🏻

**Review by langermank — APPROVED** [review]
Suggesting primitives, but not blocking

**Pull request merged**
